mlflow/store/s3_artifact_repo.py

```diff
@@ -79,6 +79,8 @@ def list_artifacts(self, path=None):
             # Objects listed directly will be files
             for obj in result.get('Contents', []):
                 file_path = obj.get("Key")
+                if file_path.endswith('/'):
```

Comment:
Are there any cases we'd get legit items that end in '/' ? Here's a potential alternative that could catch potential problems with other artifact repos as well: https://github.com/mlflow/mlflow/pull/1605. LMK what you think @antwhite - thanks!

Reply:
I can't think of a legit reason why an item would end in '/'.

In my opinion #1605 is a better fix, it is cleaner than just looking for an arbitrary suffix especially as S3 supports delimiters other than forward slash

@sueann thanks

Reply:
thanks @antwhite ! added some tests to #1605 - let me know if you have any comments.